src/DotNetLightning.Core/Channel/ChannelHelpers.fs

```diff
@@ -17,3 +24,431 @@ module internal ChannelConstantHelpers =
+
+                return
+                    transactionBuilder
+                        .AddKeys(delayedPaymentPrivKey.RawKey())
+                        .AddCoin(
+                            ScriptCoin(commitTx, uint32 toLocalIndex, toLocalScriptPubKey),
+                            CoinOptions(
+                                Sequence =
+                                    (Nullable
+                                     <| Sequence(uint32 staticChannelConfig.LocalParams.ToSelfDelay.Value))
+                            )
+                        )
+            }
+
+        let ClaimCommitTxOutputs
+            (closingTx: Transaction)
+            (staticChannelConfig: StaticChannelConfig)
+            (channelPrivKeys: ChannelPrivKeys)
+            =
+            let obscuredCommitmentNumberRes =
+                tryGetObscuredCommitmentNumber
+                    staticChannelConfig.FundingScriptCoin.Outpoint
+                    closingTx
+
+            match obscuredCommitmentNumberRes with
+            | Ok obscuredCommitmentNumber ->
+                let localChannelPubKeys = channelPrivKeys.ToChannelPubKeys()
+
+                let remoteChannelPubKeys =
+                    staticChannelConfig.RemoteChannelPubKeys
+
+                let commitmentNumber =
+                    obscuredCommitmentNumber.Unobscure
+                        staticChannelConfig.IsFunder
+                        localChannelPubKeys.PaymentBasepoint
+                        remoteChannelPubKeys.PaymentBasepoint
+
+                {
+                    MainOutput = ClaimMainOutput closingTx commitmentNumber staticChannelConfig channelPrivKeys
+                }
+            | _ ->
+                {
+                    MainOutput = Error UnknownClosingTx
+                }
+
+    module RevokedClose =
+        let private ClaimMainOutput
+            (closingTx: Transaction)
+            (commitmentNumber: CommitmentNumber)
+            (staticChannelConfig: StaticChannelConfig)
+            (remotePerCommitmentSecret: Choice<PerCommitmentSecret, PerCommitmentSecretStore>)
+            (localChannelPrivKeys: ChannelPrivKeys): Result<TransactionBuilder, OutputClaimError>
+            =
+            result {
+                let localChannelPubKeys = localChannelPrivKeys.ToChannelPubKeys()
+                let remoteChannelPubKeys = staticChannelConfig.RemoteChannelPubKeys
+
+                let! perCommitmentSecret =
+                    match remotePerCommitmentSecret with
+                    | Choice1Of2 remotePerCommitmentSecret -> Ok remotePerCommitmentSecret
+                    | Choice2Of2 remotePerCommitmentSecretStore ->
+                        let commitmentSecretOpt =
+                            remotePerCommitmentSecretStore.GetPerCommitmentSecret commitmentNumber
+
+                        match commitmentSecretOpt with
+                        | Some commitmentSecret -> Ok commitmentSecret
```

Comment:
We check for revoked closure after all our other checks return false, this makes it possible that a non-remote tx (maybe an old local commitment tx) gets caught by this function, so is it possible that this causes a False positive? (maybe there's a revoked remote commitment with the same commitment number)